core/src/main/scala/com/evolutiongaming/kafka/journal/util/GracefulFiber.scala

```diff
@@ -23,10 +23,11 @@ object GracefulFiber {
             def join = fiber.join
 
             def cancel = {
-              for {
-                cancel <- cancelRef.getAndSet(true)
-                _      <- if (cancel) ().pure[F] else fiber.joinWithNever
-              } yield {}
+              cancelRef.getAndSet(true)
+                .flatMap(cancel =>
+                  (if (cancel) ().pure[F] else fiber.joinWithNever.void)
+
+                )
```

Comment:
why replace for comprehension with `.flatMap`?

I can see usage of `Concurrent[F].when(cancel) {fiber.joinWithNever}` as possible improvement, but why structural code change?

Reply:
so I just did `Desugar for-comprehension` in IDEA and removed that last .map

Reply:
Perhaps it might compile with .void that I also put there, I'll check